dnsclient/dnsclient.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dnsclient is a client library for performing CT operations
+// over DNS.  The DNS mechanism is experimental and subject to change.
+package dnsclient
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/jsonclient"
+	"github.com/google/certificate-transparency-go/tls"
+)
+
+// DNSClient represents a DNS client for a given CT Log instance
+type DNSClient struct {
+	base     string
+	Verifier *ct.SignatureVerifier // nil for no verification (e.g. no public key available)
+	resolve  func(ctx context.Context, name string) ([]string, error)
+}
+
+// New constructs a new DNSClient instance.  The base parameter gives the
+// top-level domain name; opts can be used to provide a customer logger
```

Comment:
typo: custom

Reply:
Fixed here and at its [origin](https://github.com/google/certificate-transparency-go/commit/94d42246b1a9bfff17569e).